tests/core/test_utils.py

```diff
@@ -249,3 +251,86 @@ def test_unexpected_path_dashboard_url(self, dashboard_url):
             click.exceptions.ClickException, match="got an unexpected path"
         ):
             api_to_dashboard_url(dashboard_url)
+
+
+@patch("ggshield.core.utils.load_dotenv")
+@pytest.mark.parametrize(
+    ["cwd", "env", "expected_dotenv_path"],
+    [
+        ("sub", {}, None),
+        (".", {}, ".env"),
+        (".", {"GITGUARDIAN_DONT_LOAD_ENV": "1"}, None),
+        (".", {"GITGUARDIAN_DOTENV_PATH": ".custom-env"}, ".custom-env"),
```

Comment:
We could add a test for `"GITGUARDIAN_DOTENV_PATH": "dont_exist.txt"` that should raise an exception ?

Reply:
Good idea, going to add it. But the current behavior does not raise an exception: it prints an error and continues on its way.

I added a comment saying maybe we should not load any .env file found after that. That would change the existing behavior, but only for people who would have a GITGUARDIAN_DOTENV_PATH env var not pointing to an existing file. What do you think?

Reply:
Just went ahead and did it. I think it is what was intended after all.